### DISCUSSION TITLE
Need to manually save the file which is created with file_put_contents/filesystem.

### DISCUSSION THREAD

**Discussion by Jumborner** [symfony/symfony#42827] [Q&A] [answered]
Hi, got this weird problem. I have a function which creates translation files if they do not exist. Using file_put_contents to create the file, and adding the array with translations into it. Afterwards I do  $this->translator->trans('305304', [], null, 'es'); to check the translation, and it returns me the input ( 305304 ). 

However, if i go to the file and CTRL+S there, the translation starts working. I have tried it with filesystem `$fsObject->touch($new_file_path);
    //     $fsObject->chmod($new_file_path, 0777);
    //     $fsObject->dumpFile($new_file_path, '<?php return  ' . var_export($catalogue, true) . ';' );`

The same problem exists. I can read the file data with fread() and it would give me the result, but until I save it manually the file is not recognized by the translator.

Am i missing something?

**Comment by stof** [comment, marked as answer]
The translator component caches the loading and parsing of translation files (for performance reason). So if you modify the translation files, you need to clear your cache.

**Reply by Jumborner** [reply]
> The translator component caches the loading and parsing of translation files (for performance reason). So if you modify the translation files, you need to clear your cache.

I tried clearing the cache in a controller as described in https://symfony.com/doc/current/console/command_in_controller.html
but it does not work.

Manually running cache:clear in console however does work and the translation returns, so it does seem to be the case with the cache.
